Cookbook/Proposals/MultilineArgumentsBracket_SwiftLint_rule.md

```diff
@@ -0,0 +1,121 @@
+```swift
+func reticulateSplines(
+  spline: [Double],
+  adjustmentFactor: Double,
+  translateConstant: Int,
+  comment: String
+) -> Bool {
+  // reticulate code goes here
+}
+
+let success = reticulateSplines(
+  spline: splines,
+  adjustmentFactor: 1.3,
+  translateConstant: 2,
+  comment: "normalize the display"
+)
+```
+To reduce our time spent on fixing such nits we can use SwiftLint to warns us about such issues.
+
+## Proposed solution
+Turn on following SwiftLint rules:
+```
+- multiline_arguments
+- multiline_arguments_brackets
+- multiline_parameters
+- multiline_parameters_brackets
+```
+The documentation with samples what particular rule does can be found [here](https://github.com/realm/SwiftLint/blob/master/Rules.md#multiline-arguments).
+
+## Impact on existing codebase
+After adding rules to our codebase, Xcode reports about 2500+ violations of them. That would need to be solved before marging to develop. More than the half of the work can be automated by using SwiftFormat to automatically format files. After running [SwiftFormat](https://github.com/nicklockwood/SwiftFormat) there's around ~900 issues to be fixed manually.
+
+In my opinion the impact on the codebase will be very positive, however it will also add few changes which are not defined in our style guide. I think it's acceptable price to make it possible to resolve nits quicker which would decrease a time when a PR is open.
+
+However, the given set of rules goes a step further. `multiline_arguments_brackets` rule makes it possible achieve what we want, which is, having a trailing bracket at a function call in the new line `)`. However, this rule **always** requires to put trailing bracket `)` in a newline if the function takes more than just one line to be invoked. Even if it's only one argument.
+
+Few example what pieces of the code trigger the rule and how can this be fixed:
+
+ 1. **Trailing closure argument**:
+```swift
+AppsFlyerTracker.shared()?.continue(userActivity, restorationHandler: { restoring in
+        restorationHandler(restoring?.compactMap({ $0 as? UIUserActivityRestoring }))
+})
+```
+The fix:
+```swift
+AppsFlyerTracker.shared()?.continue(
+    userActivity,
+    restorationHandler: { restoring in
+        restorationHandler(restoring?.compactMap({ $0 as? UIUserActivityRestoring }))
+    }
+)
+```
+
+ 2. **Inlined array argument**:
+```swift
+NSLayoutConstraint.activate([
+    imageView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor, constant: -viewModel.viewSpec.secondaryLogoDistanceFromBottom),
+    imageView.centerXAnchor.constraint(equalTo: view.centerXAnchor)
+])
+```
+The fix:
+```swift
+NSLayoutConstraint.activate(
+    [
+        imageView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor, constant: -viewModel.viewSpec.secondaryLogoDistanceFromBottom),
+        imageView.centerXAnchor.constraint(equalTo: view.centerXAnchor)
+    ]
+)
+```
+or:
+```swift
+NSLayoutConstraint.activate(
```

Comment:
😢

Reply:
yes, this is also the one I don't like the most